chicago/bills.py

```diff
@@ -160,40 +86,144 @@ def scrape(self):
-                    if i == 0 :
-                        primary = True
-                        sponsorship_type = "Primary"
-                    else :
-                        primary = False
-                        sponsorship_type = "Regular"
-
-                    bill.add_sponsor(sponsor, sponsorship_type,
-                                     'person', primary)
-
-                for subject in legislation_details.get(u'Topics', []) :
-                    bill.add_subject(subject)
+                bill = Bill(identifier=legislation_summary['Record #'],
+                            legislative_session=self.session,
+                            title=title,
+                            classification=bill_type,
+                            from_organization=self.jurisdiction.name)
 
-                for attachment in legislation_details.get(u'Attachments', []) :
-                    bill.add_version_link('PDF',
-                                          attachment['url'],
-                                          mimetype="application/pdf")
+                bill.add_source(legislation_summary['url'])
 
+                bill, votes = self.addDetails(bill, legislation_summary['url'])
 
                 yield bill
+                for vote in votes :
+                    yield vote
+        
+
+
+    def extractVotes(self, action_detail_url) :
+        action_detail_page = self.lxmlize(action_detail_url)
+        vote_table = action_detail_page.xpath("//table[@id='ctl00_ContentPlaceHolder1_gridVote_ctl00']")[0]
+        votes = list(self.parseDataTable(vote_table))
+        vote_list = []
+        for vote, _, _ in votes :
+            raw_option = vote['Vote'].lower()
+            vote_list.append((VOTE_OPTIONS.get(raw_option, raw_option), 
+                              vote['Person Name']['label']))
+
+        action_detail_div = action_detail_page.xpath(".//div[@id='ctl00_ContentPlaceHolder1_pageTop1']")[0]
+        action_details = self.parseDetails(action_detail_div)
+        result = action_details['Result'].lower()
+
+        return result, vote_list
+
+
+
+    def addBillHistory(self, bill, history_table) :
+        all_votes = []
+        
+        history = self.parseDataTable(history_table)
+
+        for action, _, _ in history :
+            action_description = action['Action']
+            action_date =  action['Date'].date().isoformat()
+            try :
+                if action_description :
+                    bill.add_action(action_description,
+                                    action_date,
+                                    organization=action['Action\xa0By'],
+                                    classification=ACTION_CLASSIFICATION[action_description])
+                    if 'url' in action['Action\xa0Details'] :
+                        action_detail_url = action['Action\xa0Details']['url']
+                        result, votes = self.extractVotes(action_detail_url)
+
+                        if votes :
```

Comment:
Yes. Store the vote. I don't populate the yes and no off the roll, since it's not recorded, but I record the vote.